packages/core/utils/src/env-helper.ts

```diff
@@ -2,98 +2,127 @@ import _ from 'lodash';
 
 export type Env = typeof envFn & typeof utils;
 
-function envFn<T>(key: string, defaultValue?: T): string | T | undefined {
-  return _.has(process.env, key) ? process.env[key] : defaultValue;
+function envFn(key: string, defaultValue: string): string;
+function envFn(key: string, defaultValue?: string | null | undefined): string | undefined;
+function envFn(key: string, defaultValue?: string | null | undefined): string | undefined {
+  return _.has(process.env, key) ? process.env[key] : (defaultValue ?? undefined);
 }
 
 function getKey(key: string) {
   return process.env[key] ?? '';
 }
 
-const utils = {
-  int(key: string, defaultValue?: number): number | undefined {
-    if (!_.has(process.env, key)) {
-      return defaultValue;
-    }
+function int(key: string, defaultValue: number): number;
+function int(key: string, defaultValue?: number | null | undefined): number | undefined;
+function int(key: string, defaultValue?: number | null | undefined): number | undefined {
+  if (!_.has(process.env, key)) {
+    return defaultValue ?? undefined;
+  }
 
-    return parseInt(getKey(key), 10);
-  },
+  return parseInt(getKey(key), 10);
```

Comment:
Thought: Maybe out of scope for this PR, but `parseInt` can return `NaN`

```suggestion
  const maybeNumber = parseInt(getKey(key), 10);
  return Number.isNaN(maybeNumber) ? undefined : maybeNumber;
```